lib/chief/command.rb

```diff
@@ -1,38 +1,42 @@
 module Chief
   class Command
     def self.call(*args, &block)
-      instance = new(*args, &block)
-      result = instance.call
-      if result.is_a?(Chief::Result) && instance.send(:created_result_object)
-        result
-      else
-        fail "#{method(__method__).receiver} must call success! or fail! and return the result"
-      end
+      new(*args, &block).call
     end
 
-    def self.value(*args)
-      call(*args).value
+    def self.value(*args, &block)
+      call(*args, &block).value
     end
 
-    def call
-      fail NotImplementedError, 'Implement #call in subclass'
-    end
+    def self.inherited(subclass)
+      subclass.send(:prepend, Module.new do
+        def call
+          obj = self
+          result = super
+          if result.is_a?(Chief::Result) && obj.send(:fullfilled)
+            result
+          else
+            fail "#{obj.class}#call must call success! or fail! and return the result"
+          end
+        end
 
-    def success!(value = true)
-      create_result_object(value, nil)
-    end
+        def success!(value = true)
+          create_result_object(value, nil)
+        end
 
-    def fail!(value = false, errors = true)
-      create_result_object(value, errors)
-    end
+        def fail!(value = false, errors = true)
+          create_result_object(value, errors)
+        end
 
-    private
+        private
 
-    attr_reader :created_result_object
+        def create_result_object(value, errors)
+          @fullfilled = true
```

Comment:
super nitpick, but you have an accessor, do we want to use the setter here? (e.g. no `@`)